### ISSUE TITLE
SC2016, SC2006 complains about backticks in awscli jmespath

### ISSUE THREAD

**Issue by nitrocode** [koalaman/shellcheck#2310]
#### For bugs
- Rule Id (if any, e.g. SC1000): [SC2016](https://github.com/koalaman/shellcheck/wiki/SC2016), [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006)
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [x] The rule's wiki page does not already cover this
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

This throws SC2016 and this command works
```sh
#!/bin/bash
aws logs get-log-events \
    --query 'join(`"\n"`, events[].message)' \
    --output text
```

This throws SC2006 and this command doesn't even work...
```sh
#!/bin/bash
aws logs get-log-events \
    --query "join(`\"\n\"`, events[].message)" \
    --output text
```

#### Here's what shellcheck currently says:

`SC2016`
```
In scripts/migrate.sh line 73:
    --query 'join(`"\n"`, events[].message)' \
            ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.

For more information:
  https://www.shellcheck.net/wiki/SC2016 -- Expressions don't expand in singl...
```

`SC2006`
```
In scripts/migrate.sh line 75:
    --query "join(`\"\n\"`, events[].message)" \
                  ^------^ SC2006: Use $(...) notation instead of legacy backticked `...`.

Did you mean:
    --query "join($(\"\n\"), events[].message)" \

For more information:
  https://www.shellcheck.net/wiki/SC2006 -- Use $(...) notation instead of le...
```

#### Here's what I wanted or expected to see:

Nothing since backticks are necessary in awscli jmespath evaluation unfortunately. 

Or perhaps an alternative that would still work in the awscli command.
